hostinfo/passwdstatus.lua

```diff
@@ -0,0 +1,113 @@
+--[[
+Copyright 2014 Rackspace
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--]]
+local HostInfo = require('./base').HostInfo
+
+local forEachLimit = require('async').forEachLimit
+local fs = require('fs')
+local os = require('os')
+local spawn = require('childprocess').spawn
+local table = require('table')
+
+--[[ Passwordstatus Variables ]]--
+local Info = HostInfo:extend()
+function Info:initialize()
+  HostInfo.initialize(self)
+end
+
+function Info:run(callback)
+  if os.type() ~= 'Linux' then
+    self._error = 'Unsupported OS for passwdstatus'
+    callback()
+    return
+  end
+
+  function collectData(users)
+    --[[ Asynchronous gather user passwd info ]]--
+    local obj = {}
+
+    forEachLimit(users, 5, function(user, callback)
+      local child
+      child = spawn('passwd', {'-S', user}, {})
```

Comment:
we do have process.execFile : https://github.com/luvit/luvit/blob/master/lib/luvit/childprocess.lua#L44